pkg/externalresource/serializer.go

```diff
@@ -26,6 +43,18 @@ func (er *ExternalResource) UnmarshalYAML(unmarshal func(interface{}) error) err
 		return fmt.Errorf("there must be at least one endpoint available for the external resource")
 	}
 
+	if result.Icon == "" {
+		result.Icon = defValueForIcon
+	} else if _, ok := possibleIconValues[result.Icon]; !ok {
```

Comment:
I think this validation should rely on the backend/Kubernetes validation. If this is just some static information and doesn't have any other implication in the CLI (like handle the icon somehow), I think it shouldn't be validated in client. If we update Kubernetes definition in the CRD to allow new values, we are forcing our customers to update to the new CLI version just to be able to use that icon because their CLI might not have that icon defined yet.

We probably need to check the dev experience to see in which phase fails or how it fails (for example, not leaving inconsistency states), but maybe there is a way of checking/validating an object definition in Kubernetes without having to create the resource.

Reply:
You are right, I think we should avoid that user experience as much as possible. Relying on validation from the backend I see two points:
- Validate on CRD creation: if it does not contain the right values, it will not be created. Although it would be the easiest option to integrate as against I think that perhaps we would fail too late? it is likely that at the time we try to create the CRD fail but previously we have built images, deployed other services .. etc
- Find a mechanism to validate the CRD before starting the deployment process. (I will work on it)

Reply:
I agree that the validation must come from the backend side, but we should fail as soon as possible, without waiting for all other resources to be deployed.

If we fail after deploying the commands, compose and divert, the user may think that he didn't deploy anything and what it wasn't deployed was the external resources... 
Having to redeploy everything because of a typo it's not the best UX so validating it before deploying should be the approach to follow

Reply:
That's why I mentioned this: `checking/validating an object definition in Kubernetes without having to create the resource.` Have we checked if there is a way of doing that maybe with `client-go` library or similar?

Reply:
I'm looking for a way to do it with `client-go`